problems/06.md

```diff
@@ -2,38 +2,42 @@
 
 Great, we now how a working chat app however it is not that interesting yet. First of all we can only chat with ourself and even worse we can only see the data in the same process we are writing. Kinda boring.
 
-Since we are doing a P2P workshop we want to make P2P network connections with each other. Contrary to how we build HTTP services we don't really have a client/server model in P2P. Instead we have what is called a "swarm". A swarm is just a collection of network connections connecting us to various users all interested in a specific topic.
+Since we are doing a P2P workshop we want to make P2P network connections with each other. Contrary to how we build HTTP services we don't really have a client/server model in P2P. Instead we have what is called a "swarm". A swarm is just a collection of network sockets connecting us to various users all interested in a specific topic.
 
-To achieve this, we'll use a module called [`discovery-swarm`](https://github.com/mafintosh/discovery-swarm) which in turn uses [`discovery-channel`](https://github.com/maxogden/discovery-channel).
+To achieve this, we'll use a module called [`hyperswarm`](https://github.com/hyperswarm). This module has three strategies to connect the _swarm_ of peers:
 
-This module has three strategies to connect the _swarm_ of peers:
-
-1. Search the local network. `discovery-channel` uses a module called ['dns-discovery'](https://github.com/mafintosh/dns-discovery) that has the ability to make queries in the local network using multicast dns (MDNS).
+1. Search the local network. It makes queries in the local network using multicast dns (MDNS).
 2. Search the internet using a well-known server. The next phase is to search using dns, also with `dns-discovery`.
-3. If these methods fail, it looks in a DHT (_distributed hashtable_). For this, `discovery-channel` uses [bittorrent-dht](https://github.com/webtorrent/bittorrent-dht).
+3. If these methods fail, it looks in a DHT (_distributed hashtable_). 
 
 The third approach uses "bootstrap nodes". The bootstrap nodes
 function like known _peers_ beforehand, that we use to start to know others and
 like this form our own _swarm_. In other words, we need to know someone to enter the network. Unlike centralized services, anybody can provide these bootstrap peers, and they don't need to ever see our data.
 
-You give [discovery-swarm](https://github.com/mafintosh/discovery-swarm) a unique key that other peers using your app can use to locate each other.
+You give hyperswarm a unique key that other peers using your app can use to locate each other.
 
 ### Exercise
 
 Create a `discovery.js` file:
 
 ```js
-var discovery = require('discovery-swarm')
+var hyperswarm = require('hyperswarm')
+
+var swarm = hyperswarm()
 
-var swarm = discovery()
+var topic = Buffer.from('a49766a23610999dc5dfe05bc37cd98a9911d4b46bd25fc2cd037b9669a1e214', 'hex')
```

Comment:
It would be cool to do the hashing inline here, so folx could see that the discovery key can be derived from a human phrase.

Reply:
can you give an example? i tried to do this but it seems like it forces a particular length now.

Reply:
```js
require('crypto').createHash('sha256').update('foobar').digest()
```